components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -884,6 +884,24 @@ private void callDefaultProvisioningHandler(String username, AuthenticationConte
         // Remove role claim from local claims as roles are specifically handled.
         localClaimValues.remove(FrameworkUtils.getLocalClaimUriMappedForIdPRoleClaim(externalIdPConfig));
         localClaimValues.remove(UserCoreConstants.USER_STORE_GROUPS_CLAIM);
+
+        Map<String, String> runtimeClaims = context.getRuntimeClaims();
+        Map<String, String> remoteClaims =
+                (Map<String, String>) context.getProperty(FrameworkConstants.UNFILTERED_IDP_CLAIM_VALUES);
+
+        // Remove or revert runtime claims from local claims before calling JIT provisioning.
+        for (Map.Entry<String, String> entry : runtimeClaims.entrySet()) {
```

Comment:
Is there a possibility of getting  runtimeClaims null ?

Reply:
Since it is returned by `context.getRuntimeClaims()`, if it's not available then we'll get an empty map.